cmd/query/app/static_handler.go

```diff
@@ -126,30 +126,22 @@ func loadIndexBytes(open func(string) (http.File, error), options StaticAssetsHa
 	return indexBytes, nil
 }
 
-func (sH *StaticAssetsHandler) watch() {
-	if sH.options.UIConfigPath == "" {
-		return
-	}
-
-	watcher, err := fsnotify.NewWatcher()
-	if err != nil {
-		sH.options.Logger.Error("failed to create a new watcher for the UI config", zap.Error(err))
-		return
-	}
-
-	go func() {
-		for {
-			select {
-			case event := <-watcher.Events:
-				if event.Op&fsnotify.Remove == fsnotify.Remove {
-					// this might be related to a file inside the dir, so, just log a warn if this is about the file we care about
-					// otherwise, just ignore the event
-					if event.Name == sH.options.UIConfigPath {
-						sH.options.Logger.Warn("the UI config file has been removed, using the last known version")
-					}
-					continue
+func (sH *StaticAssetsHandler) configListener(watcher *fsnotify.Watcher) {
+	for {
+		select {
+		case event := <-watcher.Events:
+			if event.Op&fsnotify.Remove == fsnotify.Remove {
+				// this might be related to a file inside the dir, so, just log a warn if this is about the file we care about
+				// otherwise, just ignore the event
+				if event.Name == sH.options.UIConfigPath {
```

Comment:
I don't think this is correct comparison. Based on the documentation of `event.Name`, it is the name of the file without the path, whereas `UIConfigPath` may contain the path.

You probably want `filepath.Base(UIConfigPath)`

Reply:
also, this check could be done first, before checking for `event.Op`.

Reply:
The documentation said relative path: https://godoc.org/github.com/fsnotify/fsnotify#Event

Anyway, for comparison proposes I get the base of both.